api/client/src/main/java/org/projectnessie/client/http/ResponseContext.java

```diff
@@ -22,8 +22,23 @@
 /** Interface for the important parts of a response. This is created after executing the request. */
 public interface ResponseContext {
 
+  /**
+   * Get the status of the response.
+   *
+   * @deprecated use {@link #getStatus()} instead
+   */
+  @Deprecated
   Status getResponseCode() throws IOException;
```

Comment:
Could we just remove the exception declaration? If I'm not mistaken, older clients will be fine without recompilation (will have to be update only when they recompile).

Reply:
In theory, the HTTP client can be built by a pluggable `HttpClientFactory` impl loaded via service-loader. If a user has a custom implementation, we'd want it to keep working with the updated `ResponseContext` interface.

Reply:
Yes, my point is that removing a checked exception will not break binary compatibility, IIRC.

Reply:
I think it does, see [this document](https://github.com/eclipse-platform/eclipse.platform/blob/master/docs/Evolving-Java-based-APIs-2.md#Evolving-API-interfaces---API-methods) for details.

That being said, I'm happy to remove it if everybody is on board with that. I don't think anyone is actually using custom client factories in the wild.

Reply:
Okay, since it's not a binary compatibility issue, I went ahead and removed the method.

Reply:
OTOH I think the new name is better than the old one... which is a binary compatibility issue for sure. I might revert my last commit actually.

Reply:
See JLS:
```
13.4.21. Method and Constructor Throws

Changes to the throws clause of methods or constructors do not break compatibility with pre-existing binaries; these clauses are checked only at compile time. 
```

Reply:
Also indirectly via [method descriptors](https://docs.oracle.com/javase/specs/jvms/se7/html/jvms-4.html#jvms-4.3.3)

Reply:
So what should I do?

1. Keep both methods
2. Keep only `getResponseCode`, remove the `throws` clause
3. Keep only `getStatus`

Reply:
Do we actually expect custom `ResponseContext` implementations to exist?... My guess is they do not exist :)